leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -141,6 +146,9 @@ public LeadershipToken blockOnBecomingLeader() throws InterruptedException {
                     return new PaxosLeadershipToken(greatestLearned);
                 } else if (leadingStatus == StillLeadingStatus.NO_QUORUM) {
                     // If we don't have quorum we should just retry our calls.
+                    long backoffTime = (long) (noQuorumMaxDelay * Math.random());
```

Comment:
A little nitpicky: I'd probably write this differently, since if I'm not wrong this might be called on multiple threads, in which case the calls to Math.random() would contend. I'd prefer `ThreadLocalRandom.current().nextLong(0, noQuorumMaxDelay)` here.